### PR TITLE
Prac 10 implement logging

### PR DESCRIPTION
**Pull request by Andrtisar**
yup, implemented logging

### CODE REVIEW

**Review by gatoo — COMMENTED** [review]
I believe you need to add log2cpp dll too. And setup the build process to copy it to output folder after each build.
Another option is to build log4cpp as static library, not as dynamic. In this case you will not need dll at all.

File: Test/Test/ConsoleReader.cpp [hunk 1]
```diff
@@ -0,0 +1,43 @@
+#include <iostream>
+#include <conio.h>
+#include <string>
+
+#include "ConsoleReader.h"
+
+#include <log4cpp/Portability.hh>
```
Comment (gatoo) [inline Test/Test/ConsoleReader.cpp]: Do we really need all these includes?

File: Test/Test/ConsoleReader.cpp [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+#include <log4cpp/Category.hh>
+#include <log4cpp/Appender.hh>
+#include <log4cpp/FileAppender.hh>
+#include <log4cpp/OstreamAppender.hh>
+#include <log4cpp/Layout.hh>
+#include <log4cpp/BasicLayout.hh>
+#include <log4cpp/Priority.hh>
+
+int ConsoleReader::readConsole(std::string const &message) {
+	log4cpp::Category& reader =
```
Comment (gatoo) [inline Test/Test/ConsoleReader.cpp]: Why do we need to create the instance in each method we are going to have in this class? Maybe better is to have one log instance for whole class instance?

File: Test/Test/Source.cpp [hunk 3]
```diff
@@ -1,78 +1,118 @@
 #include <iostream>
 #include <fstream>
 #include <string>
+#include <stdio.h>
+
+
 #include "ConsoleReader.h"
 
+#include <log4cpp/Portability.hh>
```
Comment (gatoo) [inline Test/Test/Source.cpp]: Do we need all these includes?

File: Test/Test/Source.cpp [hunk 4]
```diff
@@ -1,78 +1,118 @@
...
+	return true;
 }
 
 
 int main() {
+	// Logging stuff
+	std::string initFileName = "log4cpp.properties";
+	log4cpp::PropertyConfigurator::configure(initFileName);
+
+	log4cpp::Category& root = log4cpp::Category::getRoot();
```
Comment (gatoo) [inline Test/Test/Source.cpp:65]: JFYI. You may use one root logger for all the application. But, sure you may have several different loggers to make it easier to configure them separately.

File: Test/Test/Source.cpp [hunk 5]
```diff
@@ -1,78 +1,118 @@
...
+
+	log4cpp::Category& main =
+		log4cpp::Category::getInstance(std::string("main"));
+
+	main.info("Started app");
+
+	/*sub2.debug("Hiding solar panels");
```
Comment (gatoo) [inline Test/Test/Source.cpp]: Do we need this commented code?

File: Test/Test/Test.vcxproj [hunk 6]
```diff
@@ -70,18 +70,21 @@
...
+    <IncludePath>lib;$(IncludePath)</IncludePath>
+  </PropertyGroup>
   <ItemDefinitionGroup Condition="'$(Configuration)|$(Platform)'=='Debug|Win32'">
     <ClCompile>
       <WarningLevel>Level3</WarningLevel>
       <Optimization>Disabled</Optimization>
       <SDLCheck>true</SDLCheck>
       <ConformanceMode>true</ConformanceMode>
-      <AdditionalIncludeDirectories>C:\Users\Andrzey\source\repos\Andrtisar\cppStuff\Test\packages\Microsoft.googletest.v140.windesktop.msvcstl.static.rt-dyn.1.8.0\build\native;C:\Users\Andrzey\source\repos\Andrtisar\cppStuff\Test\packages\Microsoft.googletest.v140.windesktop.msvcstl.static.rt-dyn.1.8.0\build\native\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
+      <AdditionalIncludeDirectories>C:\Users\Andrzey\source\repos\Andrtisar\cppStuff\Test\Test\lib;C:\Users\Andrzey\source\repos\Andrtisar\cppStuff\Test\Test\lib\log4cpp\include;%(AdditionalIncludeDirectories)</AdditionalIncludeDirectories>
```
Comment (gatoo) [inline Test/Test/Test.vcxproj]: I doubt it will compile on my PC

File: Test/log4cpp.properties [hunk 7]
```diff
@@ -0,0 +1,27 @@
+# log4cpp.properties
```
Comment (gatoo) [inline Test/log4cpp.properties]: Do we need this file twice in the project?

**Review by gatoo — CHANGES_REQUESTED** [review]

File: Test/Test/ConsoleReader.cpp (hunk 1 above)
Comment (Andrtisar) [reply Test/Test/ConsoleReader.cpp]: I guess not, it seems that I can throw away most of them

File: Test/Test/ConsoleReader.cpp (hunk 2 above)
Comment (Andrtisar) [reply Test/Test/ConsoleReader.cpp]: Yes, that makes more sense, thank you

File: Test/Test/Test.vcxproj (hunk 6 above)
Comment (Andrtisar) [reply Test/Test/Test.vcxproj]: Thanks, fixed

**Comment by Andrtisar** [comment]
> I believe you need to add log2cpp dll too. And setup the build process to copy it to output folder after each build.
Another option is to build log4cpp as static library, not as dynamic. In this case you will not need dll at all.

Uhh, is it not already a static library? It has extension .lib? Or maybe I just don't understand the question and have no idea what to do with this

**Pull request closed without merging**